### PR TITLE
Adds microk8s to run k8s locally

### PR DESCRIPTION
**Pull request by KoeSystems**
This PR adds the ability to run k8s locally

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    config.vm.provision "shell", type: "shell" do |s|
+        s.inline = "\
+            snap install microk8s --classic\n\
+            snap install docker\n\
+            microk8s.status --wait-ready\n\
+            microk8s.enable dns dashboard registry\n\
+            usermod -a -G microk8s vagrant\n\
+            echo \"alias kubectl='microk8s.kubectl'\" > /home/vagrant/.bash_aliases\n\
+            chown vagrant:vagrant /home/vagrant/.bash_aliases\n\
+            echo \"alias kubectl='microk8s.kubectl'\" > /root/.bash_aliases\n\
```
Comment (gonzaloserrano) [inline Vagrantfile:35]: so based on these aliases, you ssh into vagrant and then use `microk8s.kubectl` right?

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Vagrant.configure("2") do |config|
+    config.vm.box = "ubuntu/bionic64"
+    config.vm.hostname = "microk8s"
+    #config.vm.network "forwarded_port", id: 'http', guest: 80, host: 8080
```
Comment (gonzaloserrano) [inline Vagrantfile:4]: Excuse my ignorance, but seeing that k8s would be run inside vagrant, regarding ports:
- we expose the app ports in k8s
- then we expose the k8s ports in vagrant
- then from outside we can access them

Is that correct?

Comment (KoeSystems) [reply Vagrantfile:4]: The plan is to expose only the k8s API, then using kubectl proxy or kubectl port-forward should be enough to access any SVC deployed in k8s.

File: Vagrantfile (hunk 1 above)
Comment (KoeSystems) [reply Vagrantfile:35]: The default user in vagrant is `vagrant` so after doing `vagrant ssh` you will be able to do `kubectl` without sudo.

```
vagrant@microk8s:~$ kubectl cluster-info
Kubernetes master is running at https://127.0.0.1:16443
Heapster is running at https://127.0.0.1:16443/api/v1/namespaces/kube-system/services/heapster/proxy
CoreDNS is running at https://127.0.0.1:16443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Grafana is running at https://127.0.0.1:16443/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy
InfluxDB is running at https://127.0.0.1:16443/api/v1/namespaces/kube-system/services/monitoring-influxdb:http/proxy^
```

**Comment by KoeSystems** [comment]
@smoya You are right. There are not strong reasons to use microk8s against minikube. I traditionally used minikube and I wanted to give a go to microk8s.

I will add the minikube option too to the readme.

**Pull request merged**